### PR TITLE
Include kernel patch directly instead of by github patch url

### PR DESCRIPTION
**Pull request by jsternberg**
Includes the kernel patch as a file directly instead of relying on `fetchurl`. The patch returned by `fetchurl` changed in some manner that invalidated the hash. This method prevents a change to github's download from affecting the build.

### CODE REVIEW

**Comment by jsternberg** [comment]
Unfortunately, the reason I ran into this issue is because the previous store path had been gc'd away so I don't have the old patch. My conclusion that github must have changed something in their patch is mostly because it worked at some point so the old hash had to work and the hash now doesn't work and the only thing that could possibly change is the downloaded patch.

While I do like the idea of just changing the hash, I don't think it's necessarily reliable to assume that the patch Github sends is stable. If this happened once, it's likely to happen again. There's a certain reliability to including the kernel patch directly that's hard to beat. I've included the original hashes as variables and added a comment for how to generate the hash so it's easy for a new dev to find it.

File: pkgs/kernel/vendor.nix [hunk 1]
```diff
@@ -36,12 +36,10 @@ in
...
   kernelPatches = [{
     name = "hbiyik-panthor.patch";
-    # NOTE: This needs to be `fetchurl` instead of `fetchpatch`, because `fetchpatch`
-    # reorders the patches, and the order matters since they're generated from commits.
-    patch = fetchurl {
-      url = "https://github.com/hbiyik/linux/compare/${panthor-base}...${panthor-head}.patch";
-      hash = "sha256-/5SvlGsgHbn1i68+GASOeNZmxoZiIt280L6iUFz3MFU=";
-    };
+    # Generate using this command:
+    #   curl -o hbiyik-panthor.patch -L https://github.com/hbiyik/linux/compare/${panthor-base}...${panthor-head}.patch
```
Comment (ryan4yin) [inline pkgs/kernel/vendor.nix:40]: Could we simplify the process by eliminating the arguments `panthor-base` and `panthor-head`, and instead provide the complete URL directly?

**Comment by ryan4yin** [comment]
I’ve double-checked and the patch file in this PR is identical to the one from GitHub’s link. I’ll go ahead and merge it as soon as the little thing I mentioned is sorted out. Thanks!

**Comment by jsternberg** [comment]
I'll update the PR when I'm able to get to my personal computer a little later.

**Pull request merged**